app/main.py

```diff
@@ -1,3 +1,56 @@
+from typing import Union
+
+
 class Distance:
-    # Write your code here
-    pass
+
+    def __init__(self, km: int) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Union) -> object:
+        if isinstance(other, (int, float)):
+            return Distance(self.km + other)
+        else:
+            return Distance(self.km + other.km)
+
+    def __iadd__(self, other: Union) -> object:
+        if isinstance(other, (int, float)):
+            self.km = self + other
+            return self
+        else:
+            self.km += other.km
+            return self
+
+    def __mul__(self, number: Union) -> object:
+        return Distance(self.km * number)
+
+    def __truediv__(self, number: Union) -> object:
+        return Distance(round(self.km / number, 2))
+
+    def __lt__(self, number: Union) -> bool:
+        inst = Distance(number)
+        return self.km < inst.km
+
+    def __gt__(self, number: Union) -> bool:
```

Comment:
number it's Union?

Reply:
I thought yes, because number can be int, float or object